sdk/core/Azure.Core.Experimental/src/DynamicData.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+
+namespace Azure.Core.Dynamic
+{
+    /// <summary>
+    /// A dynamic abstraction over content data that may have an underlying
+    /// format of JSON or other format.
+    /// </summary>
+    public abstract class DynamicData
+    {
+        /// <summary>
+        /// Writes the data to the provided writer as a JSON value.
+        /// </summary>
+        /// <param name="writer">The writer to which to write the document.</param>
+        /// <param name="data">The dynamic data value to write.</param>
+        public static void WriteTo(Utf8JsonWriter writer, DynamicData data)
```

Comment:
How can this be mocked for testing?   The combination of static here and internal for the abstract method seems to signal a testability issue for customers.

If our intention is that this class should not be extended outside of Azure.Core, we should probably call that out loudly in the class-level comments.   Inevitably, someone will open an issue about this.   😄

Reply:
Oh wow, that is interesting!  I'll update the comments for now to say this type is not mockable, but we can return to it if needed either in Experimental or when we move it to Core proper.  I'm not sure I see a need to mock this type, especially since most of the APIs are internal, but I'm interested in understanding reasons we would want to.  How do we think about mocking Core types in general?  The scenarios I think about for mocking Azure SDK types are primarily around clients and models.

Reply:
I don't disagree with our intent, but based on what I've seen in issue triage - for any given public type, at some point an issue will be opened asking how to mock it.

Reply:
Cool - I added the comment to proactively address this.  Thanks!